src/IntegrationUtils.ts

```diff
@@ -322,41 +325,55 @@ function createNewIntegration(integrationFileUri: vscode.Uri, devMode? : boolean
 				console.error(error); 
 			});
 
-		let commandString = `kamel run "${absoluteRoot}"`;
+		if (extension.getDevMode() === true) {
+			// it true, always override whatever the dev mode is that's passed in
+			devMode = true;
+		}
+
+		let kamelCLI = 'kamel';
+		let kamelArgs : string[] = [];
+		kamelArgs.push('run');
+		kamelArgs.push(`${absoluteRoot}`);
 		if (devMode && devMode === true) {
-			commandString += ` --dev`;
+			kamelArgs.push('--dev');
 		}
 		if (configmap && configmap.trim().length > 0) {
-			commandString += ` --configmap=${configmap}`;
+			kamelArgs.push(`--configmap=${configmap}`);
 		}
 		if (secret && secret.trim().length > 0) {
-			commandString += ` --secret=${secret}`;
+			kamelArgs.push(`--secret=${secret}`);
 		}
 		if (resource && resource.trim().length > 0) {
 			let resourceArray = resource.split(' ');
 			if (resourceArray && resourceArray.length > 0) {
 				resourceArray.forEach(res => {
-					commandString += ` --resource="${res}"`;
+					kamelArgs.push(`--resource="${res}"`);
 				});
 			}		
 		}	
 		if (dependencyArray && dependencyArray.length > 0) {
 			dependencyArray.forEach(dependency => {
-				commandString += ` --dependency=${dependency}`;
+				kamelArgs.push(`--dependency=${dependency}`);
 			});
 		}
 		if (propertyArray && propertyArray.length > 0) {
 			propertyArray.forEach(prop => {
-				commandString += ` -p ${prop}`;
+				kamelArgs.push(`-p ${prop}`);
 			});
 		}			
-		console.log(`commandString = ${commandString}`);
+		console.log(`commandString = ${kamelCLI} ${kamelArgs}`);
 		if (devMode && devMode === true) {
 			if (extension.mainOutputChannel) {
 				extension.mainOutputChannel.show();
 			}
 		}
-		let runKubectl = child_process.exec(commandString, { cwd : foldername});
+		let runKubectl = child_process.spawn(kamelCLI, kamelArgs, { cwd : foldername});
```

Comment:
Using child_process.exec was creating a subprocess that was not getting killed when the main process was killed, so switching to spawn eliminates that issue and we have a much cleaner process that can be killed properly.

Reply:
This was the key - https://stackoverflow.com/questions/27770537/two-processes-created-by-child-process-exec